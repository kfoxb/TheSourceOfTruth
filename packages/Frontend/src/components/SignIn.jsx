import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export default function SignIn({
  error, loading, signIn, submitIfEnter, updateFormByKey,
}) {
  return (
    <Fragment>
      <p>Username: </p>
      <input onChange={updateFormByKey('username')} onKeyUp={submitIfEnter} />
      <p>Password: </p>
      <input onChange={updateFormByKey('password')} onKeyUp={submitIfEnter} type="password" />
      <button onClick={signIn}>Submit</button>
      { loading ? <p>Loading...</p> : null }
      { error ? <p>{error}</p> : null }
    </Fragment>
  );
}

SignIn.propTypes = {
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]).isRequired,
  loading: PropTypes.bool.isRequired,
  signIn: PropTypes.func.isRequired,
  submitIfEnter: PropTypes.func.isRequired,
  updateFormByKey: PropTypes.func.isRequired,
};
