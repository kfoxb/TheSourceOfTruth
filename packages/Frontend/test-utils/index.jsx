/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
