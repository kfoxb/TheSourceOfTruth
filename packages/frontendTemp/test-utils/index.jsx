/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
