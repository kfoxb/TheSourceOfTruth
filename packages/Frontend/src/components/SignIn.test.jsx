/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import SignIn from './SignIn';

describe('SignIn', () => {
  const noOp = () => {};
  const defaultProps = {
    error: false,
    loading: false,
    signIn: noOp,
    submitIfEnter: noOp,
    updateFormByKey: noOp,
  };
  const assertWithPropsToMatchSnapshot =
    createAssertWithPropsToMatchSnapshot(SignIn, defaultProps);
  it('should render', () => {
    assertWithPropsToMatchSnapshot();
  });

  it('should render an error when passed as a prop', () => {
    assertWithPropsToMatchSnapshot({ error: 'test error' });
  });

  it('should render a loading message when passed loading as a prop', () => {
    assertWithPropsToMatchSnapshot({ loading: true });
  });
});
