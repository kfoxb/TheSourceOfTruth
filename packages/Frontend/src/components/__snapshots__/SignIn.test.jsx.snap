// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render 1`] = `
<styled.div>
  <Segment>
    <Header
      as="h1"
      icon={true}
      textAlign="center"
    >
      <Icon
        as="i"
        circular={true}
        name="user"
      />
      <HeaderContent>
        Sign In
      </HeaderContent>
    </Header>
    <Form
      as="form"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Username"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Password"
        type="password"
      />
      <Button
        as="button"
        color="violet"
        fluid={true}
        onClick={[Function]}
      >
        Sign In
      </Button>
      <Divider
        horizontal={true}
      >
        Or
      </Divider>
      <Button
        as="button"
        color="blue"
        fluid={true}
      >
        Sign Up Now
      </Button>
    </Form>
  </Segment>
</styled.div>
`;

exports[`SignIn should render a loading message when passed loading as a prop 1`] = `
<styled.div>
  <Segment>
    <Header
      as="h1"
      icon={true}
      textAlign="center"
    >
      <Icon
        as="i"
        circular={true}
        name="user"
      />
      <HeaderContent>
        Sign In
      </HeaderContent>
    </Header>
    <Form
      as="form"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Username"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Password"
        type="password"
      />
      <Button
        as="button"
        color="violet"
        fluid={true}
        onClick={[Function]}
      >
        Sign In
      </Button>
      <Divider
        horizontal={true}
      >
        Or
      </Divider>
      <Button
        as="button"
        color="blue"
        fluid={true}
      >
        Sign Up Now
      </Button>
    </Form>
    <styled.div>
      <Loader
        active={true}
        inline="centered"
      />
    </styled.div>
  </Segment>
</styled.div>
`;

exports[`SignIn should render an error when passed as a prop 1`] = `
<styled.div>
  <Segment>
    <Header
      as="h1"
      icon={true}
      textAlign="center"
    >
      <Icon
        as="i"
        circular={true}
        name="user"
      />
      <HeaderContent>
        Sign In
      </HeaderContent>
    </Header>
    <Form
      as="form"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Username"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        onKeyUp={[Function]}
        placeholder="Password"
        type="password"
      />
      <Button
        as="button"
        color="violet"
        fluid={true}
        onClick={[Function]}
      >
        Sign In
      </Button>
      <Divider
        horizontal={true}
      >
        Or
      </Divider>
      <Button
        as="button"
        color="blue"
        fluid={true}
      >
        Sign Up Now
      </Button>
    </Form>
    <Message
      negative={true}
    >
      <p>
        test error
      </p>
    </Message>
  </Segment>
</styled.div>
`;
